src/lib.rs

```diff
@@ -236,6 +246,24 @@ impl<M> Context<M> {
     pub fn set_timeout(&mut self, timeout: Option<Duration>) {
         self.set_deadline(timeout.map(|t| Instant::now() + t));
     }
+
+    pub fn subscribe<E: 'static + Into<M> + Clone>(&mut self)
+    where
+        M: 'static,
+    {
+        let mut event_subscribers = self.system_handle.event_subscribers.lock();
+        let event_addr = self.myself.clone();
+
+        // TODO(bschwind) - panic if the events HashMap doesn't contain this event type ID.
+        event_subscribers.events.entry(TypeId::of::<E>()).and_modify(|subs| {
+            subs.push(Box::new(move |e| {
```

Comment:
These nested closures are a bit hard to parse visually. How about extracting the inner one into a standalone variable with a descriptive name? (Right before the call to `entry`)

Reply:
We've now removed one level of closure by using `Entry::or_default()`, so this can be hopefully considered resolved.